libraries/anvill_passes/include/anvill/Constraints.h

```diff
@@ -27,6 +27,59 @@ class Expr {
                                    const Environment &env) const = 0;
 };
 
+class Sext final : public Expr {
+ private:
+  std::unique_ptr<Expr> target;
```

Comment:
It seems like the use of `unique_ptr` here restricts you to representing trees instead of DAGs. Perhaps you want to have raw pointers, and have a vector of unique ptrs in the environment?

Reply:
This is only if there's a practical benefit.

Reply:
I don't think there is a practical benefit here. The unique ptr isnt acting as a reference to the expression, it's just there to support dynamic dispatch. If we visit an instruction two times we will just generate two std::unique_ptr<Expr>s. We are essentially generating an AST for z3.